components/admin_console/cloud_import/components/open_file.tsx

```diff
@@ -0,0 +1,362 @@
+            handleDropCb: (e: any) => handleDropCb(e),
+            handleDragOverCb,
+            handleDragOutCb,
+        },
+    );
+
+    const onOpenFileButtonClick = () => {
+        if (inputFileRef !== null && inputFileRef.current !== null) {
+            inputFileRef.current.click();
+        }
+    };
+
+    const onChangeFile = (e: any) => {
+        e.stopPropagation();
+        e.preventDefault();
+        const file = e.target.files[0];
+        onFileLoaded(file);
+    };
+
+    const onReaderLoad = (e: any) => {
+        const file = JSON.parse(e.target.result);
+        setFileValues(file);
+        setOpenFileState(FileState.COMPLETED);
+    };
+
+    const onFileLoaded = (file: any) => {
+        setOpenFileState(FileState.LOADING);
+
+        // TODO: Temporal to simulate the time loading it could take and see the screen in that state
+        setTimeout(() => {
+            const reader = new FileReader();
+            reader.onload = onReaderLoad;
+            reader.readAsText(file);
+        }, 2000);
+    };
+
+    const removeFile = () => {
+        setOpenFileState(FileState.INITIAL);
+        setDeactivateUsers(false);
+    };
+
+    return (
+        <CardContainer className='OpenFile' >
+            <FileUploadOverlay
+                overlayType='center'
+                show={showOverlay}
+            />
+            <header>
+                <FormattedMessage
+                    id='admin.cloud.import.openFile.headerStep'
+                    defaultMessage='Step 1 of 4: Select your data file'
+                />
+                <a
+                    target='_new'
+                    rel='noopener noreferrer'
+                    href={CloudLinks.HELP_IMPORTING_FROM}
+                    className='Import__GetHelp__importingFrom'
+                    onClick={() => trackEvent('cloud_admin', 'click_help_importing_from_info_pages', {screen: 'import_workspace__open_file'})}
+                >
+                    <i className='icon icon-information-outline'/>
+                    <FormattedMessage
+                        id='admin.cloud.import.openFile.helpWithMigrating'
+                        defaultMessage='Help with migrating from {chatService}'
+                        values={{chatService}}
+                    />
+                </a>
+            </header>
+            {openFileState === FileState.INITIAL && openFileStateInitialSection(dragAndDropRef, onOpenFileButtonClick, inputFileRef, onChangeFile)}
+            {openFileState === FileState.LOADING && fileStateLoadingSection()}
+            {openFileState === FileState.COMPLETED &&
+                fileStateCompletedSection(
+                    fileValues,
+                    chatService,
+                    removeFile,
+                    props.nextStepFunc,
+                    props.maxFreeUsers,
+                    isPaidTier,
+                    deactivateUsers,
+                    setDeactivateUsers)}
+        </CardContainer>
```

Comment:
Would it be more idiomatic React to make these functions `openFileStateInitialSection` and `openFileStateInitialSection` a functional component with props?